src/object/object_representation.c

```diff
@@ -1352,23 +1354,91 @@ or_put_varchar_internal (OR_BUF * buf, char *string, int charlen, int align)
   else
     {
       rc = or_put_byte (buf, 0xFF);
-
-      if (rc == NO_ERROR)
-	{
-	  OR_PUT_INT (&net_charlen, charlen);
-	  rc = or_put_data (buf, (char *) &net_charlen, OR_INT_SIZE);
-	}
+      compressable = 1;
     }
   if (rc != NO_ERROR)
     {
       return rc;
     }
 
-  /* store the string bytes */
-  rc = or_put_data (buf, string, charlen);
-  if (rc != NO_ERROR)
+  if (compressable == 1)
     {
-      return rc;
+      /* Alloc memory */
+      wrkmem = (lzo_voidp) malloc (LZO1X_1_MEM_COMPRESS);
+      if (wrkmem == NULL)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, (size_t) LZO1X_1_MEM_COMPRESS);
+	  goto cleanup;
+	}
+
+      /* Alloc memory for the compressed string */
+      compressed_string = (char *) malloc ((charlen + (charlen / 16) + 64 + 3) * sizeof (char));
+
+      /* Compress the string */
+      rc = lzo1x_1_compress (string, charlen, compressed_string, &compressed_length, wrkmem);
+
+      /* Free the working memory needed for compression */
+      if (wrkmem != NULL)
+	{
+	  free_and_init (wrkmem);
+	}
+
+      if (rc != NO_ERROR)
+	{
+	  goto cleanup;
+	}
+
+      if (compressed_length >= charlen - 8)
+	{
+	  /* Compression failed */
+	  compressed_length = 0;
+	}
+
+      /* Store the compression size */
+      assert (compressed_length < charlen - 8);
+      rc = or_put_data (buf, (char *) &compressed_length, OR_INT_SIZE);
+      if (rc != NO_ERROR)
```

Comment:
@lucianlazar2211 

Please look at the existing code:
```C
	  OR_PUT_INT (&net_charlen, charlen);
	  rc = or_put_data (buf, (char *) &net_charlen, OR_INT_SIZE);
```

I think we also need byte conversion with endian.